metadata_registry/grpc/server.py

```diff
@@ -0,0 +1,335 @@
+    session = Session(metadata_db, future=True)
+
+    # Reflect existing tables, letting sqlalchemy load linked tables where possible.
+    genome = db.Table('genome', md, autoload_with=metadata_db)
+    genome_release = db.Table('genome_release', md, autoload_with=metadata_db)
+    assembly = md.tables['assembly']
+    organism = md.tables['organism']
+    release = md.tables['ensembl_release']
+    site = md.tables['ensembl_site']
+
+    genome_select = db.select(
+            genome.c.genome_uuid,
+            organism.c.url_name,
+            organism.c.display_name,
+            organism.c.taxonomy_id,
+            organism.c.scientific_name,
+            organism.c.strain,
+            assembly.c.accession.label('assembly_accession'),
+            assembly.c.name.label('assembly_name'),
+            assembly.c.ucsc_name.label('assembly_ucsc_name'),
+            assembly.c.level.label('assembly_level'),
+            release.c.is_current
+        ).select_from(genome).join(assembly).join(organism).filter_by(
+            ensembl_name=ensembl_name
+        )
+
+    if release_version == 0:
+        genome_select = genome_select.join(genome_release).join(release).filter_by(
+            is_current=True)
+    else:
+        genome_select = genome_select.join(genome_release).join(release).filter_by(
+            version=release_version)
+
+    genome_select = genome_select.join(site).filter_by(
+        name=site_name)
+
+    genome_results = session.execute(genome_select).all()
+
+    if len(genome_results) == 1:
+        genome_data = dict(genome_results[0])
+        genome_data['ensembl_name'] = ensembl_name
+        return create_genome(genome_data)
+    else:
+        return create_genome()
+
+
+def genome_sequence_iterator(metadata_db, genome_uuid, chromosomal_only):
+    if genome_uuid is None:
+        return
+
+    # This is sqlalchemy's metadata, not Ensembl's!
+    md = db.MetaData()
+    session = Session(metadata_db, future=True)
+
+    # Reflect existing tables, letting sqlalchemy load linked tables where possible.
+    genome = db.Table('genome', md, autoload_with=metadata_db)
+    assembly = md.tables['assembly']
+    assembly_sequence = db.Table('assembly_sequence', md, autoload_with=metadata_db)
+
+    seq_select = db.select(
+            assembly_sequence.c.accession,
+            assembly_sequence.c.name,
+            assembly_sequence.c.sequence_location,
+            assembly_sequence.c.length,
+            assembly_sequence.c.chromosomal
+        ).select_from(genome).filter_by(
+            genome_uuid=genome_uuid
+        ).join(assembly).join(assembly_sequence)
+    if chromosomal_only == 1:
+        seq_select = seq_select.filter_by(chromosomal=True)
+
+    # Could batch queries, if it's too slow to get all at once;
+    # but it does not currently seem necessary.
+    seq_results = session.execute(seq_select).all()
+
+    for result in seq_results:
+        yield create_genome_sequence(dict(result))
```

Comment:
In order to interface SQLAlchemy and gRPC "Iterator Patterns" and possibly (but I'm not 100% sure) avoid to load `all()` results I think you could:

```suggestion
    for result in session.execute(seq_select):
        yield create_genome_sequence(dict(result))
```

It probably would be good to handle possible `OperationalError` exceptions here as well.

Reply:
Yes, that looks better. I take your point about catching errors, but I'm not going to bother with that right now - this only a proof of concept, so don't want to get too bogged down,